### PR TITLE
feat(compliance): add compliance field to HTML, CSV and JSON outputs including frameworks and reqs

### PR DESCRIPTION
**Pull request by MrCloudSec**
### Description

Add check compliance by default to outputs.

- convert tags to dict in JSON
- key=value in CSV
- bullet points key=value in HTML
- bullet points compliance in HTML
- Remove Description from HTML
- Merge Remediation Link in the Remediation column
- Resize HTML columns

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/lib/outputs/models.py [hunk 1]
```diff
@@ -113,22 +138,42 @@ def fill_common_data_csv(finding: dict) -> dict:
...
+                item = item_dict
+            if not unrolled_items:
+                unrolled_items = f"{item}"
+            else:
+                unrolled_items = f"{unrolled_items} {separator} {item}"
+
+    return unrolled_items
+
+
+def unroll_dict(dict: dict):
```
Comment (jfagoagas) [inline prowler/lib/outputs/models.py:199]: Do we have test for this function?

File: prowler/lib/outputs/models.py [hunk 2]
```diff
@@ -113,22 +138,42 @@ def fill_common_data_csv(finding: dict) -> dict:
...
+        "categories": unroll_list(finding.check_metadata.Categories),
+        "depends_on": unroll_list(finding.check_metadata.DependsOn),
+        "related_to": unroll_list(finding.check_metadata.RelatedTo),
         "notes": finding.check_metadata.Notes,
     }
     return data
 
 
-def __unroll_list__(listed_items: list):
+def unroll_list(listed_items: list):
```
Comment (jfagoagas) [inline prowler/lib/outputs/models.py:149]: Do we have test for this function?

File: prowler/lib/outputs/models.py [hunk 3]
```diff
@@ -11,7 +11,26 @@
 from prowler.providers.aws.lib.audit_info.models import AWS_Organizations_Info
 
 
-def generate_provider_output_csv(provider: str, finding, audit_info, mode: str, fd):
+def get_check_compliance(finding, provider, output_options):
```
Comment (jfagoagas) [inline prowler/lib/outputs/models.py:14]: Do we have test for this function?

File: prowler/lib/outputs/html.py [hunk 4]
```diff
@@ -222,13 +224,14 @@ def fill_html(file_descriptor, finding):
                 <td>{finding.region}</td>
                 <td>{finding.check_metadata.CheckTitle}</td>
                 <td>{finding.resource_id.replace("<", "&lt;").replace(">", "&gt;").replace("_", "<wbr>_")}</td>
-                <td>{str(finding.resource_tags)}</td>
+                <td>{unroll_list(finding.resource_tags)}</td>
```
Comment (jfagoagas) [inline prowler/lib/outputs/html.py]: Can we separate each element using a comma?

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/lib/outputs/html.py (hunk 4 above)
Comment (MrCloudSec) [reply prowler/lib/outputs/html.py]: We separate it with pipes (|)

Comment (jfagoagas) [reply prowler/lib/outputs/html.py]: No, the resource tags are not separated.

File: prowler/lib/outputs/models.py (hunk 3 above)
Comment (MrCloudSec) [reply prowler/lib/outputs/models.py:14]: Done

File: prowler/lib/outputs/models.py (hunk 2 above)
Comment (MrCloudSec) [reply prowler/lib/outputs/models.py:149]: Done

File: prowler/lib/outputs/models.py (hunk 1 above)
Comment (MrCloudSec) [reply prowler/lib/outputs/models.py:199]: Done

**Comment by jfagoagas** [comment]
We can't merge this PR until we fix the lateral scrolling in the HTML.

File: prowler/lib/outputs/html.py [hunk 5]
```diff
@@ -194,6 +200,7 @@ def add_html_header(file_descriptor, audit_info):
                     <th scope="col">Risk</th>
                     <th scope="col">Recomendation</th>
                     <th style="width:5%" scope="col">Recomendation URL</th>
+                    <th scope="col">Compliance</th>
```
Comment (jfagoagas) [inline prowler/lib/outputs/html.py:201]: We need to resize the "Resource Tags" column, is bigger than the title or the resource extended.

File: prowler/lib/outputs/html.py (hunk 4 above)
Comment (MrCloudSec) [reply prowler/lib/outputs/html.py]: Done

**Review by jfagoagas — CHANGES_REQUESTED** [review]

**Comment by MrCloudSec** [comment]
> We can't merge this PR until we fix the lateral scrolling in the HTML.

Done

File: prowler/lib/outputs/models.py [hunk 6]
```diff
@@ -113,26 +138,86 @@ def fill_common_data_csv(finding: dict) -> dict:
...
+                                    f"{unrolled_items} {separator} {value}="
+                                )
+                            else:
+                                unrolled_items = f"{unrolled_items}{value}"
+            elif not unrolled_items:
+                unrolled_items = f"{item}"
+            else:
+                unrolled_items = f"{unrolled_items} {separator} {item}"
+
+    return unrolled_items
```
Comment (jfagoagas) [inline prowler/lib/outputs/models.py]: I prefer to split this into two functions, one for the list and another for the tags. What do you think?

File: tests/lib/outputs/outputs_test.py [hunk 7]
```diff
@@ -201,6 +212,135 @@ def test_generate_common_csv_fields(self):
...
+
+➤PCI-3.2.1: s3
+
+➤FedRamp-Moderate-Revision-4: sc-13, sc-28
+
+➤FedRAMP-Low-Revision-4: sc-13
+"""
+        )
+
+    def test_parse_json_tags(self):
```
Comment (jfagoagas) [inline tests/lib/outputs/outputs_test.py:333]: Could you include some test to check the function's behaviour?

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/lib/outputs/html.py (hunk 5 above)
Comment (MrCloudSec) [reply prowler/lib/outputs/html.py:201]: Done

File: tests/lib/outputs/outputs_test.py (hunk 7 above)
Comment (MrCloudSec) [reply tests/lib/outputs/outputs_test.py:333]: Done

File: prowler/lib/outputs/models.py (hunk 6 above)
Comment (MrCloudSec) [reply prowler/lib/outputs/models.py]: Done

**Review by jfagoagas — APPROVED** [review]

**Pull request merged**